src/cargo/sources/registry/remote.rs

```diff
@@ -53,6 +53,10 @@ impl<'cfg> RegistryData for RemoteRegistry<'cfg> {
     }
 
     fn update_index(&mut self) -> CargoResult<()> {
+        // Ensure that this'll actually succeed later on.
+        try!(ops::http_handle(self.config));
```

Comment:
This comment and code is pretty opaque. What's 'this' and how does this commend ensure it actually succeeds?